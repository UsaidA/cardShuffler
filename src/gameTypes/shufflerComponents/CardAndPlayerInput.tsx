import { useState } from "react";
import "../../css/CardAndInputPlayer.css";

const CardAndPlayerInput: React.FC<{
  setCards: (val: any) => void;
  setPlayers: (val: any) => void;
}> = ({ setCards, setPlayers }) => {
  const [cardInput, setCardInput] = useState<number>(0);
  const [playerInput, setPlayerInput] = useState<number>(0);
  const [numOfCards, setnumOfCards] = useState<number>(52);
  const [toggle, setToggle] = useState<boolean | undefined>(false);

  const handleOnChangeCards = (val: any) => {
    const numGiven = parseInt(val.target.value);

    if(numGiven < 0 ){
      console.log("can't have negative numbers")
    }
    if (numGiven > numOfCards) {
      console.log("Can't get more cards per person");
    } else {
      console.log(numGiven, numOfCards);
      setCardInput(numGiven);
    }
  };

  const handleOnChangePlayers = (val: any) => {
    const numGiven = parseInt(val.target.value);

    if (numGiven < 0) {
      console.log("can't have negative number of players");
    } else {
      console.log(numGiven, numOfCards);
      setPlayerInput(numGiven);
    }
  };

  const onBlurOfCards = (val: any) => {
    setToggle(undefined);

    setCards(val);
  };

  const onBlurOfPlayers = (val: any) => {
    let temp = !toggle;
    setToggle(temp);

    let numOfPlayers = parseInt(val.target.value);
    let numOfCardsAvailable = numOfCards;

    numOfCardsAvailable = numOfCardsAvailable / numOfPlayers;

    setnumOfCards(numOfCardsAvailable);
    setPlayers(val);
  };

  return (
    <div className="center-container">
      <div className="input-container">
        <div className="generalDiv">
          <label htmlFor="numOfCardsInput">Number of Cards Per Player:</label>
          <input
            id="numOfCardsInput"
            value={cardInput}
            onChange={handleOnChangeCards}
            type="number"
            onBlur={onBlurOfCards}
            style={{ width: "400px", height: "50px", fontSize: "1.5em" }}
            disabled={!toggle}
          />
        </div>
        <div>
          <label htmlFor="numOfPlayersInput">Number of Players:</label>
          <input
            id="numOfPlayersInput"
            type="number"
            value={playerInput}
            onChange={handleOnChangePlayers}
            onBlur={onBlurOfPlayers}
            style={{ width: "400px", height: "50px", fontSize: "1.5em" }}
            disabled={toggle === undefined ? true : toggle}
          />
        </div>

        <br />
      </div>
    </div>
  );
};

export default CardAndPlayerInput;
