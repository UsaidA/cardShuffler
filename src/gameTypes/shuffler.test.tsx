import renderer from 'react-test-renderer';
import Shuffler from './shuffler';
import CardAndPlayerInput from './shufflerComponents/CardAndPlayerInput';
import CardPeeker from './shufflerComponents/cardPeeker';


it('Shuffler Component is rendered correctly', () => {
  const tree = renderer
    .create(<Shuffler />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('CardAndPlayerInput component rendered correctly', () => {
  const setCardsMock = jest.fn();
  const setPlayersMock = jest.fn();

  const tree = renderer
    .create(
      <CardAndPlayerInput setCards={setCardsMock} setPlayers={setPlayersMock} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

const mockTogglePeeker = jest.fn();

const mockDeck = [
  { value: 'A', suit: 'Hearts', rank: 1 },
  { value: '2', suit: 'Diamonds', rank: 2 },
  
];

it('renders correctly', () => {
  const tree = renderer
    .create(<CardPeeker deck={mockDeck} togglePeeker={mockTogglePeeker} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});