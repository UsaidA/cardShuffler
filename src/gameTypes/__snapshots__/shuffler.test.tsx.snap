// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardAndPlayerInput component rendered correctly 1`] = `
<div>
  <div
    className="input-container"
  >
    <div
      className="generalDiv"
    >
      <label
        htmlFor="numOfCardsInput"
      >
        Number of Cards Per Player:
      </label>
      <input
        disabled={true}
        id="numOfCardsInput"
        onBlur={[Function]}
        onChange={[Function]}
        style={
          Object {
            "fontSize": "1.5em",
            "height": "50px",
            "width": "400px",
          }
        }
        type="number"
        value={0}
      />
    </div>
    <div
      className="generalDiv"
    >
      <label
        htmlFor="numOfPlayersInput"
      >
        Number of Players:
      </label>
      <input
        disabled={false}
        id="numOfPlayersInput"
        onBlur={[Function]}
        onChange={[Function]}
        style={
          Object {
            "fontSize": "1.5em",
            "height": "50px",
            "width": "400px",
          }
        }
        type="number"
        value={0}
      />
    </div>
    <br />
  </div>
</div>
`;

exports[`Shuffler Component is rendered correctly 1`] = `
<div>
  <div>
    <div
      className="parent-container"
    >
      <div
        className="center-container"
      >
        <div>
          <div>
            <div
              className="input-container"
            >
              <div
                className="generalDiv"
              >
                <label
                  htmlFor="numOfCardsInput"
                >
                  Number of Cards Per Player:
                </label>
                <input
                  disabled={true}
                  id="numOfCardsInput"
                  onBlur={[Function]}
                  onChange={[Function]}
                  style={
                    Object {
                      "fontSize": "1.5em",
                      "height": "50px",
                      "width": "400px",
                    }
                  }
                  type="number"
                  value={0}
                />
              </div>
              <div
                className="generalDiv"
              >
                <label
                  htmlFor="numOfPlayersInput"
                >
                  Number of Players:
                </label>
                <input
                  disabled={false}
                  id="numOfPlayersInput"
                  onBlur={[Function]}
                  onChange={[Function]}
                  style={
                    Object {
                      "fontSize": "1.5em",
                      "height": "50px",
                      "width": "400px",
                    }
                  }
                  type="number"
                  value={0}
                />
              </div>
              <br />
            </div>
          </div>
        </div>
        <div>
          <div />
        </div>
      </div>
      <div>
        <button
          className="goButton"
          onClick={[Function]}
        >
          GO!
        </button>
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="modal-overlay"
>
  <div
    className="card-peeker-modal"
  >
    <button
      className="close-button"
      onClick={[MockFunction]}
    >
      Close
    </button>
    <button
      className="nav-button"
      onClick={[Function]}
    >
      Next
    </button>
    <div
      className="card-display"
    >
      <div />
    </div>
    <button
      className="nav-button"
      onClick={[Function]}
    >
      Previous
    </button>
  </div>
</div>
`;
